src/test/java/org/junit/internal/runners/ErrorReportingRunnerTest.java

```diff
@@ -1,27 +1,84 @@
 package org.junit.internal.runners;
 
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.InvalidTestClassError;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
 
 public class ErrorReportingRunnerTest {
-    
+
     @Test(expected = NullPointerException.class)
     public void cannotCreateWithNullClass() {
         new ErrorReportingRunner(null, new RuntimeException());
     }
-    
+
     @Test(expected = NullPointerException.class)
     public void cannotCreateWithNullClass2() {
         new ErrorReportingRunner(new RuntimeException(), (Class<?>) null);
     }
-    
+
     @Test(expected = NullPointerException.class)
     public void cannotCreateWithNullClasses() {
         new ErrorReportingRunner(new RuntimeException(), (Class<?>[]) null);
     }
-    
+
     @Test(expected = NullPointerException.class)
     public void cannotCreateWithoutClass() {
         new ErrorReportingRunner(new RuntimeException());
     }
-    
+
+    @Test
+    public void givenInvalidTestClassErrorAsCause() {
+        final List<Failure> firedFailures = new ArrayList<Failure>();
+        InvalidTestClassError testClassError = new InvalidTestClassError(TestClassWithErrors.class,
+                Arrays.asList(new Throwable("reason1"), new Throwable("reason2")));
+        ErrorReportingRunner sut = new ErrorReportingRunner(TestClassWithErrors.class, testClassError);
+
+        sut.run(new RunNotifier() {
+            @Override
+            public void fireTestFailure(Failure failure) {
+                super.fireTestFailure(failure);
+                firedFailures.add(failure);
+            }
+        });
+
+        assertThat(firedFailures.size(), is(1));
+        Throwable exception = firedFailures.get(0).getException();
+        assertThat(exception, instanceOf(InvalidTestClassError.class));
+        assertThat(((InvalidTestClassError) exception), is(testClassError));
+    }
+
+    @Test
+    public void givenInvalidTestClass_integrationTest() {
```

Comment:
Please note the integration test here. I'm not sure if this is a good place for it?